core_backend/app/llm_call/llm_prompts.py

```diff
@@ -292,29 +292,100 @@ class AlignmentScore(BaseModel):
     ).strip()
 
 
-def get_urgency_detection_prompt(condition: str, message: str) -> str:
+class UrgencyDetectionEntailment:
     """
-    Returns the prompt for the urgency detection bot.
+    Urgency detection using entailment.
     """
 
-    return textwrap.dedent(
-        (
-            """Given a [statement] and [comment], score the share of meaning
-            of [statement] covered by [comment].
-            Respond with a score between 0 and 1 with 0.1 increments.
-            """
-            """
-            Respond in json string:
+    class UrgencyDetectionEntailmentResult(BaseModel):
+        """
+        Pydantic model for the output of the urgency detection entailment task.
+        """
 
-            \{
-               statement: str
-               probability: float
-               reason: str
-            \}
-            """
-            f"""
-            statement: {condition}
-            comment: {message}
-            """
-        )
+        best_matching_rule: str
+        probability: float = Field(ge=0, le=1)
+        reason: str
+
+    _urgency_rules: List[str]
+    _prompt_base: str = textwrap.dedent(
+        """
+        You are a highly sensitive urgency detector. Score if ANY part of the
+        user message corresponds to any part of the urgency rules provided below.
+        Ignore any part of the user message that does not correspond to the rules.
+        Respond with (a) the rule that is most consistent with the user message,
+        (b) the probability between 0 and 1 with increments of 0.1 that ANY part of
+        the user message matches the rule, and (c) the reason for the probability.
+
+
+        Respond in json string:
+
+        {
+           best_matching_rule: str
+           probability: float
+           reason: str
+        }
+        """
+    ).strip()
+
+    _prompt_rules: str = textwrap.dedent(
+        """
+        Urgency Rules:
+        {urgency_rules}
+        """
     ).strip()
+
+    default_json: Dict = {
+        "best_matching_rule": "",
+        "probability": 0.0,
+        "reason": "",
+    }
+
+    def __init__(self, urgency_rules: List[str]) -> None:
+        """
+        Initialize the urgency detection entailment task with urgency rules.
+        """
+        self._urgency_rules = urgency_rules
+
+    def parse_json(self, json_str: str) -> Dict:
+        """
+        Validates the output of the urgency detection entailment task.
+        """
+
+        json_str = json_str.replace("```json", "").replace("```", "")
+        json_str = json_str.replace("\{", "{").replace("\}", "}")
```

Comment:
I created a function `core_backend.app.llm_calls.utils.remove_json_markdown` -- can we update that one and use it here?